File: Sample Apps /SampleAppWakeUp/app/src/main/java/com/mist/sample/samplewakeup/MainActivity.java
package com.mist.sample.samplewakeup;

import android.os.Bundle;
import com.mist.sample.samplewakeup.application.MainApplication;
import com.mist.sample.samplewakeup.util.AltBeaconUtil;
import org.altbeacon.beacon.BeaconManager;

public class MainActivity extends PermissionHandlerActivity {
    private static final String TAG = "SampleWakeUp";
    private static MainApplication mainApplication;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mainApplication = MainApplication.getApplication();
        checkPermissions();
        BeaconManager.setDebug(true);
    }

    @Override
    public void startBeaconMonitor() {
        AltBeaconUtil.startBeaconMonitor(mainApplication.getBeaconManager(), mainApplication);
    }
}

File: Sample Apps /Kotlin/SampleAppWakeUp/app/src/main/java/com/example/sampleappwakeup/MainActivity.kt
package com.example.sampleappwakeup

import android.os.Bundle
import com.example.sampleappwakeup.application.MainApplication
import com.example.sampleappwakeup.util.AltBeaconUtil

import org.altbeacon.beacon.BeaconManager

class MainActivity : PermissionHandlerActivity(){
    private lateinit var mainApplication : MainApplication

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        mainApplication = (application as MainApplication)
        checkPermissions()
        BeaconManager.setDebug(true)
    }

    override fun startBeaconMonitor() {
        AltBeaconUtil().startBeaconMonitor(mainApplication.getBeaconManager(), mainApplication)
    }
}